overlord/devicestate/devicemgr.go

```diff
@@ -989,6 +989,55 @@ func (m *DeviceManager) ensureSeeded() error {
 	return nil
 }
 
+var processAutoImportAssertionsImp = processAutoImportAssertions
+
+// ensureAutoImportAssertions makes sure that auto import assertions
+// get processed. assertion should be processed while seeding is in progress
+func (m *DeviceManager) ensureAutoImportAssertions() {
+	if release.OnClassic {
+		return
+	}
+
+	if m.earlyDeviceSeed == nil {
+		// we have no seed cached yet, no point to check further
+		return
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if !m.changeInFlight("seed") {
+		return
+	}
+
+	// check if we have processed auto import as already
+	var asState string
+	err := m.state.Get("auto-as-imported", &asState)
```

Comment:
s/auto-as-imported/auto-asserts-imported/

Reply:
I have changed this a bit more.
In order to capture more scenarios when system-user-assertion fails for missing serial-as, e.g. when passing system-user-assertion over REST api. I have renamed and moved place state is set to "pending".
Now "pending" state is set directly in the addUser helper when we check for serial assertion bound.
key is now "system-user-assertion"